core-rust/core-api-server/src/core_api/handlers/lts/stream_account_transaction_outcomes.rs

```diff
@@ -59,32 +60,43 @@ pub(crate) async fn handle_lts_stream_account_transaction_outcomes(
         limit,
     );
 
-    let committed_transaction_outcomes = state_versions
-        .iter()
-        .map(|state_version| {
-            Ok(to_api_lts_committed_transaction_outcome(
-                &mapping_context,
-                database
-                    .get_committed_transaction(*state_version)
-                    .expect("Transaction store corrupted"),
-                database
-                    .get_committed_transaction_receipt(*state_version)
-                    .expect("Transaction receipt index corrupted"),
-                database
-                    .get_committed_transaction_identifiers(*state_version)
-                    .expect("Transaction identifiers index corrupted"),
-            )?)
-        })
-        .collect::<Result<Vec<models::LtsCommittedTransactionOutcome>, ResponseError<()>>>()?;
-
-    Ok(models::LtsStreamAccountTransactionOutcomesResponse {
+    let mut response = models::LtsStreamAccountTransactionOutcomesResponse {
         from_state_version: to_api_state_version(from_state_version)?,
-        count: state_versions
-            .len()
-            .try_into()
-            .map_err(|_| server_error("Unexpected error mapping small usize to i32"))?,
+        count: MAX_STREAM_COUNT_PER_REQUEST as i32, // placeholder to get a better size aproximation for the header
         max_ledger_state_version: to_api_state_version(max_state_version)?,
-        committed_transaction_outcomes,
-    })
-    .map(Json)
+        committed_transaction_outcomes: Vec::new(),
+    };
+
+    // Reserve enough for the "header" fields
+    let mut current_total_size = response.get_json_size();
+    current_total_size += 8; // This should cover '[' and ']'
+    for state_version in state_versions {
+        let committed_transaction_outcome = to_api_lts_committed_transaction_outcome(
+            &mapping_context,
+            database
+                .get_committed_transaction(state_version)
+                .expect("Transaction store corrupted"),
+            database
+                .get_committed_transaction_receipt(state_version)
+                .expect("Transaction receipt index corrupted"),
+            database
+                .get_committed_transaction_identifiers(state_version)
+                .expect("Transaction identifiers index corrupted"),
+        )?;
+
+        let committed_transaction_size = committed_transaction_outcome.get_json_size();
+        if current_total_size + committed_transaction_size > MAX_STREAM_TOTAL_SIZE_PER_RESPONSE {
```

Comment:
To avoid issues where the limit is too small, I suggest we just stop processing if we're already above the limit. So it's more of a soft cap rather than a limit?